core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2432,6 +2432,26 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[util] object CallerContext {
+  private def callerContextClassExists(): Boolean = {
+    try {
+      // scalastyle:off classforname
+      Class.forName("org.apache.hadoop.ipc.CallerContext")
+      Class.forName("org.apache.hadoop.ipc.CallerContext$Builder")
+      // scalastyle:on classforname
+      true
+    } catch {
+      case _ : ClassNotFoundException => false
+    }
+  }
+
+  private def callerContextEnabled(): Boolean = {
+    SparkHadoopUtil.get.conf.getBoolean("hadoop.caller.context.enabled", false)
+  }
+
+  lazy val callerContextSupported: Boolean = callerContextEnabled && callerContextClassExists
```

Comment:
I don't think this needs to be lazy (?) and doesn't need to be broken out into methods. It can just be one `val` initialized directly upfront in one expression.